xdsl/dialects/scf.py

```diff
@@ -150,7 +150,7 @@ def parse_assignment():
         op = cls(arguments, function_type.outputs.data, before_region, after_region)
 
         if attrs is not None:
-            op.attributes = attrs.data
+            op.attributes |= attrs.data
```

Comment:
Is this related to this PR?

Reply:
Yeah, as attrs.data is now an immutabledict, this could've been `= dict(...)` but this seems cleaner